server/mentionscrawler/blueprints/user.py

```diff
@@ -0,0 +1,42 @@
+from flask import (Blueprint, current_app, jsonify, request, session, url_for)
+from werkzeug.security import generate_password_hash
+from ..models.user import MentionUser
+from sqlalchemy.exc import IntegrityError
+from psycopg2.errorcodes import UNIQUE_VIOLATION
+from ..responses import *
+
+__all__ = ["user_bp"]
+
+user_bp = Blueprint(__name__, __name__, url_prefix="/")
+
+
+@user_bp.route("/user", methods=["POST"])
+def register():
+    if request.is_json:
+        body = request.get_json()
+        if body["email"] and body["name"] and body["password"]:
+            new_user = MentionUser(body["email"], body["name"], generate_password_hash(body["password"]))
+            try:
+                MentionUser.commit_user(new_user)
+            except IntegrityError as e:
+                print(e)
+                if e.orig.pgcode == UNIQUE_VIOLATION:
+                    return bad_request_response(new_user.email + " already taken")
+            except Exception as e:
+                print(e)  # Will eventually replace this with a logger that prints to a file
+                return error_response("Something went wrong and we don't know what!")
```

Comment:
I think it's better not catch this and instead add an error handler to intercept internal server errors and return a user-friendly message https://flask.palletsprojects.com/en/1.1.x/patterns/errorpages/

Reply:
Wouldn't this be helpful though because we can log what the actual error is? And then generate an error page on the front end?

Reply:
On the handler, you can get the original error and log it before returning the response

Reply:
alright I'll implement that